test/func.js

```diff
@@ -521,4 +554,585 @@ describe('func', function () {
     });
   });
 
-});
\ No newline at end of file
+  it('count works with happner-datastore style parameter',  function(done) {
+
+    const dataItem = [
+      {data: {"test": "data1"}},
+      {data: {"test": "data1"}},
+      {data: {"test": "data1"}},
+      {data: {"test": "data2"}},
+      {data: {"test": "data2"}}
+    ]
+
+    let insertCount = 0
+
+    async function countEntries()
+    {
+
+      let countPromise = util.promisify(serviceInstance.count).bind(serviceInstance);
+      let countAll = await countPromise('/countTest/num*');
+      expect(countAll).to.be(5);
+      let count1 = await countPromise('/countTest/num1');
+      expect(count1).to.be(1);
+      let count2 = await countPromise('/countTest/*');
+      expect(count2).to.be(5);
+      done()
+
+
+
+    }
+
+
+    for(let i = 0 ; i < dataItem.length ; ++i)
+    {
+      serviceInstance.upsert(`/countTest/num${i}`,dataItem[i],{},false,(err)=>{
+        if(err)
+          done(err)
+        insertCount++
+        if(insertCount === dataItem.length )
+        {
+          countEntries()
+        }
+      });
+
+
+    }
+
+
+
+
+
+  });
+
+  it('find ',  function(done) {
+
+    const dataItem = [
+      {data: {"test": "data1"}},
+      {data: {"test": "data1"}},
+      {data: {"test": "data1"}},
+      {data: {"test": "data2"}},
+      {data: {"test": "data2"}}
+    ]
+
+    let insertCount = 0
+
+    async function findEntries()
+    {
+
+      // serviceInstance.findNew("/findTest/num1",{criteria : {"query":"data.test:data1"}},(err,data)=>{
+      //   debugger
+      //   done();
+      // })
+
+      // const body = {
+      //   'query': {
+      //
+      //         'query_string': {
+      //           "query": "(data.test:data1)"
+      //         }
+      //
+      //   }
+      // };
+
+
+      const body = {
+        'query': {
+          'constant_score': {
+            'filter': {
+              "query_string": {
+                "query":     [{'path' :'/findTest/num1' }],
+              }
+            }
+          }
+        }
+      };
+
+      serviceInstance.find("/findTest/num1",body,(err,data)=>{
+
+        done();
+      })
+
+    }
+
+
+    for(let i = 0 ; i < dataItem.length ; ++i)
+    {
+      serviceInstance.upsert(`/findTest/num${i}`,dataItem[i],{},false,(err)=>{
+        if(err)
+          done(err)
```

Comment:
return to avoid done called more than once

Reply:
fixed